include/swift/AST/DiagnosticsSema.def

```diff
@@ -3952,8 +3952,8 @@ ERROR(attr_incompatible_with_objc,none,
       (DeclAttribute, DescriptiveDeclKind))
 
 ERROR(final_not_on_accessors,none,
-      "'final' cannot be applied to accessors, it must be put on the "
-      "%select{var|let|subscript}0", (unsigned))
+      "only %select{var|let|subscript}0 itself can be marked as 'final'"
+      ,(unsigned))
```

Comment:
The are several issues with this diagnostic:
1. As you noted earlier, a parsing error is getting in its way.
1. Consequently, there are no tests for it.
1. A `let` cannot be `final`.

I will open an issue for 1 & 2, don’t bother addressing them in this PR. Suggestion for 3:
* ```suggestion
      "only the %select{property|subscript}0 itself can be marked as 'final'"
      ,(bool))
  ```
* Update emission site accordingly.

Reply:
done, but do we know why we previously supported let ? sounds odd to me
d00dd6ca27cd3de3e75245ef20a6d7eaa6565899

Reply:
> but do we know why we previously supported let 

We never did. The inclusion of `let` in the message was probably an oversight.

Reply:
> I will open an issue for 1 & 2

https://github.com/swiftlang/swift/issues/75933.